extension/src/paymentHandler/payment-handler.js

```diff
@@ -139,6 +141,39 @@ function _validatePaymentRequest(paymentObject, authToken) {
   return null
 }
 
+function _isOpenInvoiceMethodType(makePaymentRequestObj) {
+  const addCommercetoolsLineItemsFlag = _getAddCommercetoolsLineItemsFlag(
+    makePaymentRequestObj
+  )
+  if (addCommercetoolsLineItemsFlag !== undefined) {
+    return addCommercetoolsLineItemsFlag
+  }
+
+  if (_isKlarna(makePaymentRequestObj) || _isAffirm(makePaymentRequestObj))
+    return true
+
+  const addCommercetoolsLineItemsAppConfigFlag =
+    config.getModuleConfig().addCommercetoolsLineItems
+  if (addCommercetoolsLineItemsAppConfigFlag === true)
+    return _isPaymentMethodRequiresLineItems(makePaymentRequestObj)
+
+  return false
+}
+
+function _getAddCommercetoolsLineItemsFlag(makePaymentRequestObj) {
+  let addCommercetoolsLineItems
+  if ('addCommercetoolsLineItems' in makePaymentRequestObj) {
+    if (
+      makePaymentRequestObj.addCommercetoolsLineItems === true ||
+      makePaymentRequestObj.addCommercetoolsLineItems === false
+    ) {
+      addCommercetoolsLineItems =
+        makePaymentRequestObj.addCommercetoolsLineItems
+    }
+  }
+  return addCommercetoolsLineItems
+}
```

Comment:
This whole method could be maybe shorter:

```
function _isCommercetoolsLineItemsFlagTrue(makePaymentRequestObj) {
  if (makePaymentRequestObj['addCommercetoolsLineItems'] === true) {
    return true
  return false
}
```

Reply:
the idea was to pick if it's set to true/false not only a check for true case, if it's undefined it will go with other cases.

Reply:
Pair review: add a comment for the fallback to omit the future overrides like above as it seems more straightforward.

Reply:
https://github.com/commercetools/commercetools-adyen-integration/pull/918/commits/99c22db3ac4add36c995448008801321afbebb20